### DISCUSSION TITLE
Using Next.js middleware and withAuth breaks Playwright E2E tests with Prisma adapter in Github actions only

### DISCUSSION THREAD

**Discussion by juan-carlos-correa** [nextauthjs/next-auth#9854] [Help]
I'm working in a [personal open-source project](https://github.com/product-makers-hub/open-micro-saas) using next-auth as part of the tech stack.

The issue that I have is that if I add a middleware using the `withAuth` from next-auth, my tests are failing in Github actions only (in local everything is fine). 

Here is when I had the test errors:

- [Commit change](https://github.com/product-makers-hub/open-micro-saas/commit/054d10f9b91ee33d06040ddce5efc6b10aa3b4aa).
- [GA log errors](https://github.com/product-makers-hub/open-micro-saas/actions/runs/7676361402/job/20987085300).

The middleware with withAuth:

```
import { withAuth } from "next-auth/middleware";
import { NextResponse } from "next/server";

import { authConfig } from "./config";
import { ADMIN_ROLE_NAME } from "./consts/roles-consts";

const adminRoutes = [
  "/admin",
  "/admin/dashboard",
  "/admin/dashboard/user-management",
];

export default withAuth(
  // `withAuth` augments your `Request` with the user's token.
  function middleware(req) {
    const isAdminRoute = adminRoutes.includes(req.nextUrl.pathname);

    const isUserAdmin =
      (req.nextauth.token?.role as any)?.name === ADMIN_ROLE_NAME;

    if (isAdminRoute && !isUserAdmin) {
      return NextResponse.redirect(
        new URL(authConfig.normalUserCallbackUrl, req.url),
      );
    }
  },
  {
    callbacks: {
      authorized: ({ token }) => {
        return token?.isActive as any;
      },
    },
    pages: {
      signIn: authConfig.loginUrl,
    },
  },
);

/**
 * The config for the middleware, it needs to be static.
 * Therefore, it's not possible to use variables like adminRoutes array
 */
export const config = {
  matcher: [
    "/admin/",
    "/admin/dashboard",
    "/admin/dashboard/user-management",
    "/profile",
    "/dashboard",
  ],
};
```

I downloaded the Playwright html report and after debugging and some checks, I found out that in the test, it's behaving like the DB is empty (which is weird since there are another tests where the user is required to be authenticated).

The next SS is how the UI looks like when I'm getting the users from the DB, but in this case, behaves like there are not users (but I have an user logged in in the app).

<img width="431" alt="image" src="https://github.com/nextauthjs/next-auth/assets/10777626/36ce93f8-e337-4124-9edb-96719d7b2901">

As proof, I reverted the middleware and it's working as expected.

This is how it looks like when is working:

<img width="646" alt="image" src="https://github.com/nextauthjs/next-auth/assets/10777626/887fa22d-789b-40bd-977c-68ad248cfe30">

- [Commit change without the middleware](https://github.com/product-makers-hub/open-micro-saas/commit/d491c2f6d29a9a7286ba360f8d48aa716372bae9).
- [GA log without error](https://github.com/product-makers-hub/open-micro-saas/actions/runs/7702822283/job/20991907683).

**The curious part is that Prisma is not returning data from the DB when the tests are failing, as if the DB does not have data in the users table.**

And if I just use the Next.js middlware with a simple console log (without the `withAuth` wrapper from next-auth), the tests are passing:

- [Commit with Next.js middleware without the `withAuth` that is passing](https://github.com/product-makers-hub/open-micro-saas/commit/401372ad21fbde7fe78dcab426adbc70f7d2a0a5).

What does this means? 

If I use the `withAuth`, my tests fail, and if I don't, my tests pass. That's why I believe is a possible issue with `withAuth` middleware.

My first guess was that I was doing something wrong in my tests or I was missing something in the GA configurations, but is weird that only when I add a middleware I have the error and without the middleware I'm not having it, isn't? 

I would appreciate any help or suggest 🙏

**Comment by VintorezVS** [comment]
Seems like I have the same issue. Do you find any solution to this?
